### PR TITLE
feat(helm): support pulling envoyGateway image from a private registry

### PR DESCRIPTION
**Pull request by shahar-h**
**What this PR does / why we need it**:
Add support for pulling envoyGateway image from a private registry by setting `deployment.envoyGateway.imagePullSecrets` in helm chart values.
Sample use case: pulling image from an internal mirror.

**Which issue(s) this PR fixes**: N/A

### CODE REVIEW

File: charts/gateway-helm/templates/certgen.yaml [hunk 1]
```diff
@@ -33,6 +33,10 @@ spec:
           value: {{ .Values.kubernetesClusterDomain }}
         image: {{ .Values.deployment.envoyGateway.image.repository }}:{{ .Values.deployment.envoyGateway.image.tag | default .Chart.AppVersion }}
         imagePullPolicy: {{ .Values.deployment.envoyGateway.imagePullPolicy }}
+        {{- with .Values.deployment.envoyGateway.imagePullSecrets }}
+        imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
```
Comment (cnvergence) [inline charts/gateway-helm/templates/certgen.yaml:38]:
```suggestion
          {{- toYaml . | nindent 8 }}
```

**Review by cnvergence — APPROVED** [review]
lgtm, just small nit

File: charts/gateway-helm/templates/certgen.yaml (hunk 1 above)
Comment (shahar-h) [reply charts/gateway-helm/templates/certgen.yaml]: thanks, fixed

**Comment by shahar-h** [comment]
`coverage-test` failed with:
```log
--- FAIL: TestProcessHTTPRoutes (0.00s)
    --- FAIL: TestProcessHTTPRoutes/httproute_with_one_filter_from_extension (0.00s)
        testing.go:1465: race detected during execution of test
    testing.go:1465: race detected during execution of test
```
It doesn't look related to my change.
How can I retrigger it?

**Comment by shahar-h** [comment]
Can this be merged if there is no additional feedback.

**Review by Xunzhuo — APPROVED** [review]

**Comment by shahar-h** [comment]
Can this be merged?

**Review by zirain — APPROVED** [review]

**Pull request merged**